packages/features/ee/round-robin/roundRobinManualReassignment.ts

```diff
@@ -0,0 +1,433 @@
+import { getBookerBaseUrl } from "@calcom/lib/getBookerUrl/server";
+import logger from "@calcom/lib/logger";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import { BookingReferenceRepository } from "@calcom/lib/server/repository/bookingReference";
+import { getTimeFormatStringFromUserTimeFormat } from "@calcom/lib/timeFormat";
+import { prisma } from "@calcom/prisma";
+import { WorkflowActions, WorkflowMethods, WorkflowTriggerEvents } from "@calcom/prisma/enums";
+import { userMetadata as userMetadataSchema } from "@calcom/prisma/zod-utils";
+import type { EventTypeMetadata } from "@calcom/prisma/zod-utils";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+
+import type { BookingSelectResult } from "./utils/bookingSelect";
+import { bookingSelect } from "./utils/bookingSelect";
+import { getDestinationCalendar } from "./utils/getDestinationCalendar";
+import { getTeamMembers } from "./utils/getTeamMembers";
+
+enum ErrorCode {
+  InvalidRoundRobinHost = "invalid_round_robin_host",
+  UserIsFixed = "user_is_round_robin_fixed",
+}
+
+export const roundRobinManualReassignment = async ({
+  bookingId,
+  newUserId,
+  orgId,
+}: {
+  bookingId: number;
+  newUserId: number;
+  orgId: number | null;
+}) => {
+  const roundRobinReassignLogger = logger.getSubLogger({
+    prefix: ["roundRobinManualReassign", `${bookingId}`],
+  });
+
+  let booking = await prisma.booking.findUnique({
+    where: { id: bookingId },
+    select: bookingSelect,
+  });
+
+  if (!booking || !booking.user) {
+    roundRobinReassignLogger.error(`Booking ${bookingId} not found or has no associated user`);
+    throw new Error("Booking not found or has no associated user");
+  }
+
+  const eventTypeId = booking.eventTypeId;
+  if (!eventTypeId) {
+    roundRobinReassignLogger.error(`Booking ${bookingId} does not have an event type id`);
+    throw new Error("Event type not found");
+  }
+
+  const eventType = await getEventTypesFromDB(eventTypeId);
+  if (!eventType) {
+    roundRobinReassignLogger.error(`Event type ${eventTypeId} not found`);
+    throw new Error("Event type not found");
+  }
+
+  const eventTypeHosts = eventType.hosts.length
+    ? eventType.hosts
+    : eventType.users.map((user) => ({
+        user,
+        isFixed: false,
+        priority: 2,
+        weight: 100,
+        weightAdjustment: 0,
+        schedule: null,
+      }));
+
+  const fixedHost = eventTypeHosts.find((host) => host.isFixed);
+  const currentRRHost = booking.attendees.find((attendee) =>
+    eventTypeHosts.some((host) => !host.isFixed && host.user.email === attendee.email)
+  );
+  const newUserHost = eventTypeHosts.find((host) => host.user.id === newUserId);
+
+  if (!newUserHost) {
+    throw new Error(ErrorCode.InvalidRoundRobinHost);
+  }
+
+  if (newUserHost.isFixed) {
+    throw new Error(ErrorCode.UserIsFixed);
+  }
+
+  const originalOrganizer = booking.user;
+  const hasOrganizerChanged = !fixedHost && booking.userId !== newUserId;
```

Comment:
```suggestion
  const hasOrganizerChanged = !fixedHost;
```
Wouldn't this already be enough? If there is no fixed host then organizer would always change as their is always only one RR host assigned